### PR TITLE
Add GitHub Actions workflows

### PR DESCRIPTION
**Pull request by jameswklassen**
## Description
The Travis CI flow was no longer running.

## In this PR
- Replace Travis CI workflow with GitHub actions
- Update linting framework
- Fix tests (have been broken since [#30](https://github.com/Brightspace/user-profile-behavior/pull/30))

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: CI
+on: pull_request
+jobs:
+  test:
+    name: Lint/Test
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
```
Comment (EvanLemieux) [inline .github/workflows/ci.yml:7]: I think we want this now to be `[self-hosted, Linux, AWS]`

Comment (jameswklassen) [reply .github/workflows/ci.yml:7]: This is a public repo, so it has to be `ubuntu-latest`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+      - name: Checkout
+        uses: Brightspace/third-party-actions@actions/checkout
+      - name: Set up node
+        uses: Brightspace/third-party-actions@actions/setup-node
+        with:
+          node-version: 14
+          registry-url: ${{secrets.ARTIFACTORY_NPM_REGISTRY}}
+          scope: '@d2l'
```
[the remark this answers is not in the document]
Comment (jameswklassen) [reply .github/workflows/ci.yml:16]: In order to publish this to CodeArtifact, the repo has to be private.

Once Gaudi starts making public repos private, I think we'll add CodeArtifact stuff.

[the remark this answers is not in the document]
Comment (EvanLemieux) [reply .github/workflows/ci.yml:16]: Got it, don't mind me then

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: Brightspace/third-party-actions@actions/checkout
+      - name: Set up node
+        uses: Brightspace/third-party-actions@actions/setup-node
+        with:
+          node-version: 14
+          registry-url: ${{secrets.ARTIFACTORY_NPM_REGISTRY}}
```
Comment (jadasc) [inline .github/workflows/ci.yml:15]: I don't think this repo has this secret

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+name: Release
+on:
+  push:
+    branches:
+      - main
+jobs:
+  release:
+    if: "!contains(github.event.head_commit.message, 'skip ci')"
```
[the remark this answers is not in the document]
Comment (jameswklassen) [reply .github/workflows/release.yml:8]: Oh yeah right, this repo seems like a good place to test it

File: .github/workflows/ci.yml (hunk 3 above)
Comment (jadasc) [reply .github/workflows/ci.yml:15]: (I'm not sure where/how to configure this repo to get these secrets)

Comment (jameswklassen) [reply .github/workflows/ci.yml:15]: I believe that secret is only available to private repositories.

Comment (jadasc) [reply .github/workflows/ci.yml]: Ah okay, so not needed here then 👍

**Review by jadasc — APPROVED** [review]

**Pull request merged**